crates/sui-json-rpc/src/api/mod.rs

```
@@ -46,3 +48,12 @@ pub fn cap_page_limit(limit: Option<usize>) -> usize {
         limit
     }
 }
+
+pub fn cap_page_objects_limit(limit: Option<usize>) -> usize {
+    let limit = limit.unwrap_or_default();
+    if limit > MAX_GET_OWNED_OBJECT_SIZE || limit == 0 {
+        MAX_GET_OWNED_OBJECT_SIZE
+    } else {
+        limit
+    }
+}
```

nit: I would prefer to return an error rather than silent cap the limit